front/src/applications/editor/tools/rangeEdition/components/RangeEditionTrackRangeList.tsx

```diff
@@ -30,48 +33,43 @@ const TrackRangesList = () => {
   const ranges = entity.properties.track_ranges || [];
   const displayedRanges = showAll ? ranges : ranges.slice(0, DEFAULT_DISPLAYED_RANGES_COUNT);
 
-  const rangesByRoute = groupBy(displayedRanges, (range) => {
-    const isCurrentRange = compareTrackRange(range);
-    // eslint-disable-next-line no-restricted-syntax
-    for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
-      if (isCurrentRange(routeExtra[routeKey]) || trackRanges.some(isCurrentRange)) {
-        return routeKey;
-      }
+  function makeFullTrackList() {
+    if (withRouteName) {
+      const rangesByRoute = groupBy(displayedRanges, (range) => {
+        const isCurrentRange = compareTrackRange(range);
+        // eslint-disable-next-line no-restricted-syntax
+        for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
+          if (isCurrentRange(routeExtra[routeKey]) || trackRanges.some(isCurrentRange)) {
```

Comment:
Why not use isEqual ?
```suggestion
      const rangesByRoute = groupBy(displayedRanges, (currentRange) => {
        // eslint-disable-next-line no-restricted-syntax
        for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
          if (isEqual(currentRange, routeExtra[routeKey]) ||
        trackRanges.some((range) => isEqual(currentRange, range))) {
```

Reply:
it’s more verbose. 
`trackRanges.some(isCurrentRange)` reads like a sentence

Reply:
See my comment below, can you modify partialIsEqual to be a common function and not a partially applied one ?

Reply:
done